src/bridger/Bridger.sol

```diff
@@ -0,0 +1,201 @@
+
+    /**
+     * @dev Permit the spender to spend the specified amount of tokens on behalf of the owner
+     * @param asset address of the token
+     * @param amount amount of tokens
+     * @param expiresAt deadline for the signature
+     * @param signature signature to be recovered
+     */
+    function _permit(address asset, uint256 amount, uint256 expiresAt, bytes memory signature) private {
+        // fix Index range access is only supported for dynamic calldata arrays.
+        (uint8 v, bytes32 r, bytes32 s) = abi.decode(signature, (uint8, bytes32, bytes32));
+        ERC20Permit(asset).permit(msg.sender, address(this), amount, expiresAt, v, r, s);
+    }
+
+    /**
+     * @dev Deposit the specified amount of tokens
+     * @param asset address of the token
+     * @param amount amount of tokens
+     */
+    function _deposit(address asset, uint256 amount) private {
+        require(amount > 0, "Bridger: amount must be greater than 0");
+        require(IERC20(asset).balanceOf(msg.sender) >= amount, "Bridger: insufficient balance");
+        require(IERC20(asset).allowance(msg.sender, address(this)) >= amount, "Bridger: insufficient allowance");
+        IERC20(asset).transferFrom(msg.sender, address(this), amount);
+        deposits[msg.sender][asset] += amount;
+    }
+
+    /**
+     * @dev Swaps ERC20->ERC20 tokens held by this contract using a 0x-API quote.
+     */
+    function _fillQuote(
+        // The `sellTokenAddress` field from the API response.
+        IERC20 sellToken,
+        // The `buyTokenAddress` field from the API response.
+        IERC20 buyToken,
+        // The `allowanceTarget` field from the API response.
+        address spender,
+        // The `to` field from the API response.
+        address payable swapTarget,
+        // The `data` field from the API response.
+        bytes calldata swapCallData
+    ) private returns (uint256) {
+        // Checks that the swapTarget is actually the address of 0x ExchangeProxy
+        // require(swapTarget == exchangeProxy, "Target not ExchangeProxy");
+
+        // Track our balance of the buyToken to determine how much we've bought.
+        uint256 boughtAmount = buyToken.balanceOf(address(this));
+
+        // Give `spender` an infinite allowance to spend this contract's `sellToken`.
+        // Note that for some tokens (e.g., USDT, KNC), you must first reset any existing
+        // allowance to 0 before being able to update it.
+        require(sellToken.approve(spender, type(uint256).max), "Failed to approve spender");
+        // Call the encoded swap function call on the contract at `swapTarget`,
+        // passing along any ETH attached to this function call to cover protocol fees.
+        (bool success,) = swapTarget.call{value: msg.value}(swapCallData);
+        require(success, "SWAP_CALL_FAILED");
+        // Keep the protocol fee redunds
+        // msg.sender.transfer(address(this).balance);
```

Comment:
i think this should be uncommented, right?

Reply:
no, I think we want to keep them because we are going to pay for their gas fees anyway

Reply:
its also a potential security issue to send ETH away